src/PrestaShopBundle/Controller/Admin/Sell/Catalog/ManufacturerController.php

```diff
@@ -0,0 +1,355 @@
+    ) {
+        $manufacturerGridFactory = $this->get('prestashop.core.grid.grid_factory.manufacturer');
+        $manufacturerGrid = $manufacturerGridFactory->getGrid($manufacturerFilters);
+
+        $manufacturerAddressFactory = $this->get('prestashop.core.grid.grid_factory.manufacturer_address');
+        $manufacturerAddressGrid = $manufacturerAddressFactory->getGrid($manufacturerAddressFilters);
+
+        return $this->render('@PrestaShop/Admin/Sell/Catalog/Manufacturer/index.html.twig', [
+            'enableSidebar' => true,
+            'help_link' => $this->generateSidebarLink($request->attributes->get('_legacy_controller')),
+            'manufacturer_grid' => $this->presentGrid($manufacturerGrid),
+            'manufacturer_address_grid' => $this->presentGrid($manufacturerAddressGrid),
+        ]);
+    }
+
+    /**
+     * Provides filters functionality
+     *
+     * @param Request $request
+     *
+     * @return RedirectResponse
+     */
+    public function searchAction(Request $request)
+    {
+        $definitionFactory = $this->get('prestashop.core.grid.definition.factory.manufacturer_grid_definition_factory');
+        $definitionFactory = $definitionFactory->getDefinition();
+
+        $gridFilterFormFactory = $this->get('prestashop.core.grid.filter.form_factory');
+        $searchParametersForm = $gridFilterFormFactory->create($definitionFactory);
+        $searchParametersForm->handleRequest($request);
+
+        $filters = [];
+        if ($searchParametersForm->isSubmitted()) {
+            $filters = $searchParametersForm->getData();
+        }
+
+        return $this->redirectToRoute('admin_manufacturers_index', ['filters' => $filters]);
+    }
+
+    public function createManufacturerAction()
+    {
+        //todo: implement
+        return $this->redirectToRoute('admin_manufacturers_index');
+    }
+
+    public function viewManufacturerAction()
+    {
+        //todo: implement
+        return $this->redirectToRoute('admin_manufacturers_index');
+    }
+
+    public function editManufacturerAction()
+    {
+        //todo: implement
+        return $this->redirectToRoute('admin_manufacturers_index');
+    }
+
+    /**
+     * Deletes manufacturer
+     *
+     * @param $manufacturerId
+     *
+     * @return RedirectResponse
+     */
+    public function deleteAction($manufacturerId)
+    {
+        try {
+            $this->getCommandBus()->handle(new DeleteManufacturerCommand((int) $manufacturerId));
+            $this->addFlash(
+                'success',
+                $this->trans('Successful deletion.', 'Admin.Notifications.Success')
+            );
+        } catch (DomainException $e) {
+            $this->addFlash('error', $this->getErrorMessageForException($e, $this->getErrorMessages()));
+        }
+
+        return $this->redirectToRoute('admin_manufacturers_index');
+    }
+
+    /**
+     * Deletes manufacturers on bulk action
+     *
+     * @param Request $request
```

Comment:
Don't forget `@AdminSecurity` and `@DemoRestricted` annotation ;)

Reply:
Veeeeery important indeed 😄